addons/stock/models/stock_move.py

```diff
@@ -866,15 +867,14 @@ def _onchange_lot_ids(self):
             ('move_id', '!=', self._origin.id),
             ('state', '!=', 'cancel')
         ])
-
         counter = self.env['stock.move.line'].search_count([
             ('company_id', '=', self.company_id.id),
             ('product_id', '=', self.product_id.id),
             ('move_id', '=', self._origin.id),
             ('lot_id', '=', False),
             ('lot_name', '!=', False),
         ])
-        self.update({'quantity_done': len(self.lot_ids) + counter})
+        self.update({'quantity_done': (len(self.lot_ids) + counter)*self.product_uom.factor})
```

Comment:
You should rather transform each move line to move UoM. with _compute_quantity and sum the result.

Reply:
check _quantity_done_compute in stock_move.py